Add `plot()` functionality to audio metrics

shhs29 opened the pull request:
## What does this PR do?

Fixes #1406

Currently, plot functionality is added for pesq metric. The test for plotting passes locally. However, the workflows on CI are failing.

## Before submitting

- [x] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/metrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to **update the docs**?
- [x] Did you write any new **necessary tests**?

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

Borda commented:
@SkafteNicki @justusschock mind have look :otter:

SkafteNicki commented:
Hi @shhs29, I looked through your PR and fixed the problems with the tests (a missing installation in the CI was the main problem). Can you add for the remaining audio metrics and then we can get this merged? :)
Thanks for your work.

shhs29 commented:
> Hi @shhs29, I looked through your PR and fixed the problems with the tests (a missing installation in the CI was the main problem). Can you add for the remaining audio metrics and then we can get this merged? :) Thanks for your work.

@SkafteNicki Thanks a lot for the help. I will add plot functionality for other audio metrics and make it available soon.

SkafteNicki commented:
> > Hi @shhs29, I looked through your PR and fixed the problems with the tests (a missing installation in the CI was the main problem). Can you add for the remaining audio metrics and then we can get this merged? :) Thanks for your work.
> 
> @SkafteNicki Thanks a lot for the help. I will add plot functionality for other audio metrics and make it available soon.

No problem. Just tag me when you are done with the rest and I can help cleanup and fix anything that may be missing/fails :]

shhs29 commented:
> > > Hi @shhs29, I looked through your PR and fixed the problems with the tests (a missing installation in the CI was the main problem). Can you add for the remaining audio metrics and then we can get this merged? :) Thanks for your work.
> > 
> > 
> > @SkafteNicki Thanks a lot for the help. I will add plot functionality for other audio metrics and make it available soon.
> 
> No problem. Just tag me when you are done with the rest and I can help cleanup and fix anything that may be missing/fails :]

@SkafteNicki I have added plot functionality for all the audio metrics. However, I am unsure about the lower and upper bounds of these metrics. Also, a couple of Github actions are failing, which I suspect is due to some incorrect/insufficient imports. I would appreciate your feedback on the same.

SkafteNicki commented on src/torchmetrics/audio/stoi.py:
```diff
@@ -11,16 +11,20 @@
 __doctest_requires__ = {"ShortTimeObjectiveIntelligibility": ["pystoi"]}
 
+if not _MATPLOTLIB_AVAILABLE:
+    __doctest_skip__ = ["SignalNoiseRatio.plot", "ScaleInvariantSignalNoiseRatio.plot"]
```
```suggestion
    __doctest_skip__ = ["ShortTimeObjectiveIntelligibility.plot"]
```

SkafteNicki commented:
@shhs29 thanks, I will take a look at it soon. We are experience problems with our CI at the moment, which should hopefully be solved soon also.

Borda commented on src/torchmetrics/audio/pesq.py:
```diff
@@ -119,3 +124,50 @@ def update(self, preds: Tensor, target: Tensor) -> None:
+                If no value is provided, will automatically call `metric.compute` and plot that result.
+            ax: An matplotlib axis object. If provided will add plot to that axis
+
+        Returns:
+            fig: Figure object
+            ax: Axes object
```
```suggestion
        Returns:
            Figure and Axes object
```
